ls1intum/Artemis
# `Notifications`: Fix incorrect mail subjects

### Checklist
#### General


#### Server

### Motivation and Context


There was a bug where the email for some notification types did not have the correct subject but a placeholder.
### Description

For some notification types, the mail subject was not generated based on the user language and notification type, but was just taken from the notification title. This means previously to https://github.com/ls1intum/Artemis/pull/6106 the subject always was the English text and after https://github.com/ls1intum/Artemis/pull/6106 it was always set to a placeholder value. I added a German and English translation in case there was none. So now the mail subjects are displayed correctly and are translated.


### Steps for Testing

Prerequisites:
- Test on a test server that has set up a mail service (ts0 or ts1)
- Log in as a user whose emails you can access (your tum online account)
- 1 Instructor

1. Log in to Artemis
2. Click on your username -> settings -> notifications
![Screenshot 2023-04-17 at 17-39-31 Notification Settings](https://user-images.githubusercontent.com/44734881/232538676-e26c99ad-48cb-465f-9a95-690c245e3787.png)
3. Activate the email notifications you want to test
4. Do the action to trigger the notification (e.g. release an exercise)
5. Check your email and verify that the email subject is set correctly

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage




No changes.

### Screenshots


After:
<img width="1308" alt="Bildschirmfoto 2023-04-17 um 17 34 21" src="https://user-images.githubusercontent.com/44734881/232537436-ca7f90a4-f5f4-405c-b682-346c7aeda387.png">

Before:
<img width="1308" alt="Bildschirmfoto 2023-04-17 um 17 34 45" src="https://user-images.githubusercontent.com/44734881/232537519-0f74303c-27e2-4155-af3f-4ab19f68087a.png">
